R/session-cookie.R

```diff
@@ -1,25 +1,115 @@
 #' Store session data in encrypted cookies.
-#' @param key The secret key to use. This must be consistent across all sessions
-#'   where you want to save/restore encrypted cookies. It should be a long and
+#'
+#' \code{plumber} uses the crypto R package \code{sodium}, to encrypt/decrypt
+#' \code{req$session} information for each server request.
+#'
+#' The cookie's secret encryption \code{key} value must be consistent to maintain
+#' \code{req$session} information between server restarts.
+#'
+#' @section Storing secure keys:
+#' While it is very quick to get started with user session cookies using
+#' \code{plumber}, please exercise precaution when storing secure key information.
+#' If a malicious person were to gain access to the secret \code{key}, they woud
+#' be able to eavesdrop on all \code{req$session} information and/or tamper with
+#' \code{req$session} information being processed.
+#'
+#' Please: \itemize{
+#' \item Do NOT store keys in source control.
+#' \item Do NOT store keys on disk with permissions that allow it to be accessed by everyone.
+#' \item Do NOT store keys in databases which can be queried by everyone.
+#' }
+#'
+#' Instead, please: \itemize{
+#' \item Use a key management system, such as
+#' \href{https://github.com/r-lib/keyring}{'keyring'} (preferred)
+#' \item (or) Store it on a disk that only you have access to.
+#' Such as modifying the file permissions to "user read only"
+#' (\code{chmod 400 myfile.txt}) to your file from being modified and prevent
+#' others from reading your file.
```

Comment:
"on a disk that only you have access to", that sounds like a USB key or something.

The next sentence is awkwardly worded as well.

How about:

> (or) Store the secret in a file on disk with appropriately secure permissions, such as "user read/write only" (\code{Sys.chmod("myfile.txt", mode = "0600")}), to prevent others from reading it.

Reply:
Should this be 400 or 600 permissions? (read only vs read/write).  

My thought is to safe guard against the password accidentally changing (or changing because of a program run by the user) by removing write permissions.

Reply:
If 600, I'll change the example as well

Reply:
I can see both sides but 400 seems like overkill to me. There's a much larger risk in any of your .R files changing accidentally but we don't encourage anyone to set 400 on those. And I thought there might be a chance that some people wouldn't actually know how to open permissions up if they do want to edit or delete that file.